frontend/src/core/cells/cells.ts

```diff
@@ -733,6 +733,20 @@ export function notebookIsRunning(state: NotebookState) {
   );
 }
 
+export function notebookScrollToRunning(state: NotebookState) {
+  // find cell that is currently in "running" state
+  const { cellIds } = store.get(notebookAtom);
+  const cellIdx = Object.values(state.cellRuntime).findIndex(
+    (cell) => cell.status === "running",
+  );
```

Comment:
i think you can avoid wiring the state through, and might be able to change this to:

```tsx
  const { cellRuntime } = store.get(notebookAtom);
  const cell = Objects.entries(cellRuntime).find(
    ([cellId, runtime]) => runtime.status === "running",
  )
  if (!cell) {
     return;
  }
  const view = getCellEditorView(cell[0]);
  view?.dispatch({
    selection: { anchor: 0 },
    effects: [EditorView.scrollIntoView(0, { y: "center" })],
  });
```

Reply:
Looks good, thanks for the suggestion :) I'll give it a try.

Reply:
I updated the PR to your suggestion. Thanks for the improvement :)